flink/src/test/java/org/apache/iceberg/flink/TestFlinkCatalogTable.java

```diff
@@ -239,6 +241,29 @@ public void testCreatePartitionTable() throws TableNotExistException {
     Assert.assertEquals(Collections.singletonList("dt"), catalogTable.getPartitionKeys());
   }
 
+  @Test
+  public void testCreateTableWithFormatV2ThroughTableProperty() throws Exception {
+    sql("CREATE TABLE tl(id BIGINT) WITH ('format-version'='2')");
+
+    Table table = table("tl");
+    Assert.assertEquals("should create table using format v2",
+        2, ((BaseTable) table).operations().current().formatVersion());
+  }
+
+  @Test
+  public void testUpgradeTableWithFormatV2ThroughTableProperty() throws Exception {
```

Comment:
It would be nice to have a test that validates you can't downgrade a table using the table property.